### PR TITLE
Support custom source sets

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
 1. Adds the ability to handle a custom source set in JS code generation.
 2. Bumps dependency on `tool-base` and `config`.

Notable changes:
  * `GenerateJsonParsers` action class introduced. The class encapsulates the code generation logic which was previously implemented as static methods under `McJsPlugin`.
  * `McJsExtension` was renamed to `McJsOptions`.
  * All path-related properties of the plugin options were removed (similarly to those in `McJavaOptions`).

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, during my absence, if needed, please feel free to pick this PR up. Otherwise, I hope to pick it up on Monday next week.

File: build.gradle.kts [hunk 1]
```diff
@@ -157,24 +153,24 @@ subprojects {
     JavadocConfig.applyTo(project)
     CheckStyleConfig.applyTo(project)
 
+    val javaVersion = 8
```
Comment (dmdashenkov) [inline build.gradle.kts:156]: Please use the constant instead of a literal. Detekt is not happy.

**Review by dmdashenkov — APPROVED** [review]
LGTM with a small note.

File: build.gradle.kts (hunk 1 above)
Comment (armiol) [reply build.gradle.kts:156]: As discussed, there is no good solution on converting between `JavaVersion.VERSION_1_8` and `int 8`, therefore, this warning will be suppressed.

**Pull request merged**